windows/src/main/csharp/ch/cyberduck/ui/ViewModels/TransferViewModel.cs

```diff
@@ -0,0 +1,244 @@
+            Transfer = transfer;
+
+            transferTypeName = CultureInfo.CurrentCulture.TextInfo.ToTitleCase(Transfer.Model.getType().name());
+
+            var progressState = Observable.Create<TransferProgressModel>(observer =>
+            {
+                bool emitEmpty = true;
+                var subscription = controller.Progress.Watch(transfer.Model)
+                    .Subscribe(next =>
+                    {
+                        if (next.Reason == ChangeReason.Remove)
+                        {
+                            observer.OnNext(default);
+                        }
+                        else
+                        {
+                            observer.OnNext(next.Current);
+                        }
+
+                        emitEmpty = false;
+                    }, observer.OnError, observer.OnCompleted);
+
+                if (emitEmpty)
+                {
+                    observer.OnNext(null);
+                    if (subscription == Disposable.Empty)
+                    {
+                        observer.OnCompleted();
+                    }
+                }
+
+                return subscription;
+            }).Subscribe(v => ProgressState = v);
+
+            Transfer.Roots.Connect().ObserveOnDispatcher()
+                .Transform(m => new TransferItemViewModel(m))
+                .DisposeMany().Bind(out roots).Subscribe();
+
+#pragma warning disable CS8509 
+            TransferDirection = transfer.Model switch
+#pragma warning restore CS8509 
+            {
+                DownloadTransfer => TransferDirection.Download,
+                UploadTransfer or CopyTransfer => TransferDirection.Upload,
+                SyncTransfer => TransferDirection.Sync,
+            };
+
+            transfer.PropertyChanged += OnTransferChanged;
+        }
+
+        public void Cancel()
+        {
+            ProgressState?.Cancel();
+        }
+
+        void IDisposable.Dispose()
+        {
+            Transfer.PropertyChanged -= OnTransferChanged;
+            progressStateNotifications.Dispose();
+        }
+
+        partial void OnProgressStateChanged(TransferProgressModel value)
+        {
+            if (value is null)
+            {
+                Running = false;
+            }
+            else
+            {
+                Running = true;
+                (ProgressPending, Progress) = value.Progress switch
+                {
+                    null => (true, 0),
+                    double progress => (false, progress)
+                };
+
+                OnPropertyChanged(nameof(ItemStatus));
+                OnPropertyChanged(nameof(MessageText));
+                OnPropertyChanged(nameof(ProgressText));
+            }
+        }
+
+        partial void OnProgressStateChanged(TransferProgressModel old, TransferProgressModel value)
+        {
+            progressStateNotifications.Disposable = value?.WhenAnyPropertyChanged().Subscribe(OnProgressStateChanged);
```

Comment:
SerialDisposable disposes previous disposable, then assigns new value.
This way we don't need to track the event-subscription manually.